crypto/bio/bio_lib.c

```diff
@@ -848,35 +858,42 @@ int BIO_connect_retry(BIO *bio, int timeout)
         return 0;
     }
 
-    if (!blocking)
-        BIO_set_nbio(bio, 1);
+    if (polling_milliseconds < 0)
+        polling_milliseconds = 100;
+    BIO_set_nbio(bio, !blocking);
 
- retry: /* it does not help here to set SSL_MODE_AUTO_RETRY */
-    rv = BIO_do_handshake(bio); /* This indirectly calls ERR_clear_error(); */
+ retry: /* it does not always help to set SSL_MODE_AUTO_RETRY */
```

Comment:
The comment mentions SSL_MODE_AUTO_RETRY
This error handling is for BIO_TYPE_CONNECT,
not for BIO_TYPE_SSL, right?

Reply:
I also think that SSL_MODE_AUTO_RETRY is specifically for SSL-type BIOs, and this is one of the reasons why in general it is not sufficient.

Reply:
I just want to make sure this comment is not understood that this logic here is
prepared to handle BIO_TYPE_SSL, those have other special wait conditions,
for instance BIO_get_retry_reason() == BIO_RR_SSL_X509_LOOKUP

Reply:
Do you have a suggestion how to phrase the comment better,
or should I simply drop it because it may give more confusion than explanation?

Reply:
Yes please drop the comment.

Reply:
Done